provider/cluster/kube/client.go

```diff
@@ -122,8 +122,7 @@ func (c *client) Deployments(ctx context.Context) ([]cluster.Deployment, error)
 		return nil, err
 	}
 
-	var deployments []cluster.Deployment
-
+	deployments := make([]cluster.Deployment, len(manifests.Items))
```

Comment:
this should use `make([]cluster.Deployment, 0, len(manifests.Items))`

Reply:
good catch, thanks!